### ISSUE TITLE
Potential false positive: CVE-2022-42919

### ISSUE THREAD

**Issue by mlathara** [aquasecurity/trivy#4175]
## Description

I'm not sure if this is a trivy issue or a AVD issue. In trying to fix CVE-2022-42919, I installed python 3.9.16 in my image and would've expected a clean scan based on the description of the issue both on [AVD](https://avd.aquasec.com/nvd/2022/cve-2022-42919/) and [NVD](https://nvd.nist.gov/vuln/detail/CVE-2022-42919). Excerpt from NVD:
```
cpe:2.3:a:python:python:*:*:*:*:*:*:*:*   Show Matching CPE(s) | From (including)3.9.0 | Up to (excluding)3.9.16
```

## What did you expect to happen?

I expected a clean scan since I installed python3.9.16

## What happened instead?

Trivy flagged the 3.9.16 with CVE-2022-42919

## Output of run with `--debug`:

```
2023-05-04T09:51:46.690-0700    DEBUG   Severities: ["HIGH" "CRITICAL"]
2023-05-04T09:51:46.696-0700    DEBUG   cache dir:  /nfs/home/mlathara/.cache/trivy
2023-05-04T09:51:46.696-0700    DEBUG   DB update was skipped because the local DB is the latest
2023-05-04T09:51:46.697-0700    DEBUG   DB Schema: 2, UpdatedAt: 2023-05-04 12:07:41.217245832 +0000 UTC, NextUpdate: 2023-05-04 18:07:41.217245332 +0000 UTC, DownloadedAt: 2023-05-04 16:17:51.775921021 +0000 UTC
2023-05-04T09:51:46.699-0700    INFO    Vulnerability scanning is enabled
2023-05-04T09:51:46.699-0700    DEBUG   Vulnerability type:  [os library]
2023-05-04T09:51:46.699-0700    INFO    Secret scanning is enabled
2023-05-04T09:51:46.699-0700    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-05-04T09:51:46.699-0700    INFO    Please see also https://aquasecurity.github.io/trivy/v0.41/docs/secret/scanning/#recommendation for faster secret detection
2023-05-04T09:51:46.707-0700    DEBUG   No secret config detected: trivy-secret.yaml
2023-05-04T09:51:46.707-0700    DEBUG   Image ID: sha256:dfc740f2fa2447c9f48ae54bdcb9685e1fc6a47ecfec6c8593822f9ee3e20d55
2023-05-04T09:51:46.707-0700    DEBUG   Diff IDs: [sha256:bf8cedc62fb3ef98ad0dff2be56ca451dd3ea69abd0031ad3e0fe5d9f9e4dfff sha256:c9b1512ecfd9dbc9608b48156b8f1befd3bf42a6547d503e076259c24a8947e9 sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef sha256:f8ecc151f27e971a72fe078cb384640a30d180036c6ca889983cbbf8e58450a3 sha256:e69093e97239055c9eaba6366e29fe2180b0e2998f2ffef783d96d31319de15b sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef sha256:4148da694a6d01ebb9380fd772063b2eba10872a28e4f21c87935fab03286999 sha256:6adf6b6260e0cc645623841b17280eff4af4af053e2b7f05c6fb3e4a3c8ccb19 sha256:62019b606184272a1a65255c074db1ca15eab11cb8231548939a2a87c22ee4a0 sha256:059792b43aec5331aafef796aa0d03b9690e87232c551ee97dea5c94eebc3b2f sha256:cc985d9efb12d8aac97543d5597470283ae078c8ad92342fb67ca8d77df8242d]
2023-05-04T09:51:46.707-0700    DEBUG   Base Layers: [sha256:bf8cedc62fb3ef98ad0dff2be56ca451dd3ea69abd0031ad3e0fe5d9f9e4dfff]
2023-05-04T09:51:46.758-0700    INFO    Detected OS: ubuntu
2023-05-04T09:51:46.758-0700    INFO    Detecting Ubuntu vulnerabilities...
2023-05-04T09:51:46.758-0700    DEBUG   ubuntu: os version: 20.04
2023-05-04T09:51:46.758-0700    DEBUG   ubuntu: the number of packages: 263
2023-05-04T09:51:46.792-0700    INFO    Number of language-specific files: 1
2023-05-04T09:51:46.792-0700    INFO    Detecting python-pkg vulnerabilities...
2023-05-04T09:51:46.792-0700    DEBUG   Detecting library vulnerabilities, type: python-pkg, path: 
2023-05-04T09:51:46.871-0700    DEBUG   Found an ignore file .trivyignore
2023-05-04T09:51:46.872-0700    DEBUG   These IDs will be ignored: []
2023-05-04T09:51:46.872-0700    DEBUG   Found an ignore file .trivyignore
2023-05-04T09:51:46.872-0700    DEBUG   These IDs will be ignored: []
{
  "SchemaVersion": 2,
  "ArtifactName": "image:latest",
  "ArtifactType": "container_image",
  "Metadata": {
    "OS": {
      "Family": "ubuntu",
      "Name": "20.04"
    },
    "ImageID": "sha256:dfc740f2fa2447c9f48ae54bdcb9685e1fc6a47ecfec6c8593822f9ee3e20d55",
    "DiffIDs": [
      "sha256:bf8cedc62fb3ef98ad0dff2be56ca451dd3ea69abd0031ad3e0fe5d9f9e4dfff",
      "sha256:c9b1512ecfd9dbc9608b48156b8f1befd3bf42a6547d503e076259c24a8947e9",
      "sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef",
      "sha256:f8ecc151f27e971a72fe078cb384640a30d180036c6ca889983cbbf8e58450a3",
      "sha256:e69093e97239055c9eaba6366e29fe2180b0e2998f2ffef783d96d31319de15b",
      "sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef",
      "sha256:4148da694a6d01ebb9380fd772063b2eba10872a28e4f21c87935fab03286999",
      "sha256:6adf6b6260e0cc645623841b17280eff4af4af053e2b7f05c6fb3e4a3c8ccb19",
      "sha256:62019b606184272a1a65255c074db1ca15eab11cb8231548939a2a87c22ee4a0",
      "sha256:059792b43aec5331aafef796aa0d03b9690e87232c551ee97dea5c94eebc3b2f",
      "sha256:cc985d9efb12d8aac97543d5597470283ae078c8ad92342fb67ca8d77df8242d"
    ],
    "RepoTags": [
      "image:latest"
    ],
<omitting image info>
  },
  "Results": [
    {
      "Target": "image:latest (ubuntu 20.04)",
      "Class": "os-pkgs",
      "Type": "ubuntu",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2022-42919",
          "PkgID": "libpython3.9-minimal@3.9.16-1+focal1",
          "PkgName": "libpython3.9-minimal",
          "InstalledVersion": "3.9.16-1+focal1",
          "Layer": {
            "DiffID": "sha256:059792b43aec5331aafef796aa0d03b9690e87232c551ee97dea5c94eebc3b2f"
          },
          "SeveritySource": "ubuntu",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2022-42919",
          "DataSource": {
            "ID": "ubuntu",
            "Name": "Ubuntu CVE Tracker",
            "URL": "https://git.launchpad.net/ubuntu-cve-tracker"
          },
          "Title": "python: local privilege escalation via the multiprocessing forkserver start method",
          "Description": "Python 3.9.x before 3.9.16 and 3.10.x before 3.10.9 on Linux allows local privilege escalation in a non-default configuration. The Python multiprocessing library, when used with the forkserver start method on Linux, allows pickles to be deserialized from any user in the same machine local network namespace, which in many system configurations means any user on the same machine. Pickles can execute arbitrary code. Thus, this allows for local user privilege escalation to the user that any forkserver process is running as. Setting multiprocessing.util.abstract_sockets_supported to False is a workaround. The forkserver start method for multiprocessing is not the default start method. This issue is Linux specific because only Linux supports abstract namespace sockets. CPython before 3.9 does not make use of Linux abstract namespace sockets by default. Support for users manually specifying an abstract namespace socket was added as a bugfix in 3.7.8 and 3.8.3, but users would need to make specific uncommon API calls in order to do that in CPython before 3.9.",
          "Severity": "HIGH",
          "CVSS": {
            "nvd": {
              "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 7.8
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H",
              "V3Score": 7.8
            }
          },
          "References": [
            "https://access.redhat.com/errata/RHSA-2022:8493",
            "https://access.redhat.com/security/cve/CVE-2022-42919",
            "https://bugzilla.redhat.com/2138705",
            "https://bugzilla.redhat.com/show_bug.cgi?id=2138705",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-42919",
            "https://errata.almalinux.org/9/ALSA-2022-8493.html",
            "https://errata.rockylinux.org/RLSA-2022:8493",
            "https://github.com/python/cpython/compare/v3.10.8...v3.10.9",
            "https://github.com/python/cpython/compare/v3.9.15...v3.9.16",
            "https://github.com/python/cpython/issues/97514",
            "https://github.com/python/cpython/issues/97514#issuecomment-1310277840",
            "https://linux.oracle.com/cve/CVE-2022-42919.html",
            "https://linux.oracle.com/errata/ELSA-2022-8493.html",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FKGCQPIVHEAIJ77R3RSNSQWYBUDVWDKU/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/P2LHWWEI5OBQ6RELULMVU6KMDYG4WZXH/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PI5DYIED6U26BGX5IRZWNCP6TY4M2ZGZ/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/QLUGZSEAO3MBWGKCUSMKQIRYJZKJCIOB/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/R6KGIRHSENZ4QAB234Z36HVIDTRJ3MFI/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RDK3ZZBRYFO47ET3N4BNTKVXN47U6ICY/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VCRKBB5Y5EWTJUNC7LK665WO64DDXSTN/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/XX6LLAXGZVZ327REY6MDZRMMP47LJ53P/",
            "https://nvd.nist.gov/vuln/detail/CVE-2022-42919",
            "https://python-security.readthedocs.io/vuln/multiprocessing-abstract-socket.html",
            "https://security.gentoo.org/glsa/202305-02",
            "https://security.netapp.com/advisory/ntap-20221209-0006/",
            "https://ubuntu.com/security/notices/USN-5713-1",
            "https://ubuntu.com/security/notices/USN-5888-1",
            "https://www.cve.org/CVERecord?id=CVE-2022-42919"
          ],
          "PublishedDate": "2022-11-07T00:15:00Z",
          "LastModifiedDate": "2023-05-03T11:15:00Z"
        },
...
```

## Output of `trivy -v`:

```
Version: 0.41.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-05-04 12:07:41.217245832 +0000 UTC
  NextUpdate: 2023-05-04 18:07:41.217245332 +0000 UTC
  DownloadedAt: 2023-05-04 16:17:51.775921021 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-04-19 01:06:36.379637155 +0000 UTC
  NextUpdate: 2023-04-22 01:06:36.379636755 +0000 UTC
  DownloadedAt: 2023-04-19 23:24:09.971565555 +0000 UTC
Policy Bundle:
  Digest: sha256:19a017cdc798631ad42f6f4dce823d77b2989128f0e1a7f9bc83ae3c59024edd
  DownloadedAt: 2023-03-04 01:31:40.337424959 +0000 UTC
```

## Additional details (base image name, container registry info...):
N/A

**Comment by DmitriyLewen** [comment]
Hello @mlathara 
Thanks for your report!

We use vendor OS advisories for os packages. Read more about that [here](https://aquasecurity.github.io/trivy/v0.41/docs/scanner/vulnerability/os/#data-source-selection). 

Ubuntu curreently doesn't have fixed version for `python 3.9` for `Focal Ubuntu` - https://ubuntu.com/security/CVE-2022-42919.
When Ubuntu will add fixed version - we will not get this CVE.

Regards, Dmitriy

**Comment by mlathara** [comment]
Ah, interesting. I had used deadsnakes PPA to get a more recent python to try and avoid the CVE...but from the sounds of it, even though its not a official Ubuntu package, you'll use OS advisories for it. Thanks for the clarification.
